tests/unitary/src/communication/test_communicator_manager.cpp

```diff
@@ -135,3 +137,101 @@ TEST_CASE( "register the same communicator backend twice", "[communicator_manage
     REQUIRE( backends.size() == 1 );
     REQUIRE( backends[0] == name );
 }
+
+TEST_CASE( "get_preferred_backend should return the best available backend", "[communicator_manager]" ) 
+{
+    auto mock1 = std::make_unique<mock_communicator_backend>();
+    const std::string name1 = "mock1";
+    REQUIRE_CALL(*mock1, get_name())
+        .RETURN(name1)
+        .TIMES(2);
+    REQUIRE_CALL(*mock1, get_priority())
+        .RETURN(backend_priority::normal)
+        .TIMES(AT_LEAST(1));
```

Comment:
at least needed?

Reply:
Yes, we call std algorithms which may not be consistent across implementations of the standard library.